test/goose/specs_test.clj

```diff
@@ -10,14 +10,24 @@
     [goose.test-utils :as tu]
     [goose.worker :as w]
 
-    [clojure.test :refer [deftest is are]])
+    [clojure.test :refer [deftest is are]]
+    [tech.v3.dataset :as ds])
   (:import
     (clojure.lang ExceptionInfo)
     (java.time Instant)))
 
 (defn single-arity-fn [_] "dummy")
 (def now (Instant/now))
 
+;;; A tech.v3.DS was reported to have mis-matching types post
+;;; de/serialization. This ugliness is the ONLY way to reproduce this bug.
+(def tech-v3-dataset-sample
```

Comment:
> This ugliness is the ONLY way to reproduce this bug

suggestion: it is unclear from this what exactly the test checks for, can we add that to the comments? And include details about what the `tech.v3.DS` functions do, so that it is straightforward to reason about the test.

Reply:
Added comments to explain the test.